hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/FieldResolverTest.java

```diff
@@ -0,0 +1,232 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.connector.mongodb;
+
+import com.hazelcast.jet.sql.impl.connector.mongodb.FieldResolver.DocumentField;
+import com.hazelcast.sql.impl.schema.MappingField;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.CreateCollectionOptions;
+import com.mongodb.client.model.ValidationOptions;
+import org.bson.BsonDocument;
+import org.bson.BsonType;
+import org.bson.Document;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.MongoDBContainer;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.sql.impl.type.QueryDataType.INT;
+import static com.hazelcast.sql.impl.type.QueryDataType.VARCHAR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
+public class FieldResolverTest {
+    static final String TEST_MONGO_VERSION = System.getProperty("test.mongo.version", "6.0.3");
+    @ClassRule
+    public static MongoDBContainer mongoContainer = new MongoDBContainer("mongo:" + TEST_MONGO_VERSION);
+
+    @Test
+    public void testResolvesFieldsViaSchema() {
+        try (MongoClient client = MongoClients.create(mongoContainer.getConnectionString())) {
+            String databaseName = "testDatabase";
+            String collectionName = "people";
+            MongoDatabase testDatabase = client.getDatabase(databaseName);
+
+            CreateCollectionOptions options = new CreateCollectionOptions();
+            ValidationOptions validationOptions = new ValidationOptions();
+            validationOptions.validator(BsonDocument.parse(
+                    "{\n" +
+                    "    $jsonSchema: {\n" +
+                            "      bsonType: \"object\",\n" +
+                            "      title: \"Person Object Validation\",\n" +
+                            "      required: [ \"firstName\", \"lastName\", \"birthYear\" ],\n" +
+                            "      properties: {" +
+                            "        \"firstName\": { \"bsonType\": \"string\" }\n" +
+                            "        \"lastName\": { \"bsonType\": \"string\" }\n" +
+                            "        \"birthYear\": { \"bsonType\": \"int\" }\n" +
+                            "      }\n" +
+                            "    }\n" +
+                            "  }\n"
+            ));
```

Comment:
That's actually pretty cool we support this, if you are already making notes for a blog post, this is one to highlight!